packages/arb-token-bridge-ui/src/components/TransferPanel/TransferPanel.tsx

```diff
@@ -475,6 +481,12 @@ export function TransferPanel() {
                   payload: false
                 })
                 setTransferring(false)
+                if (
+                  !isSmartContractWallet &&
+                  isFathomNetworkName(l2NetworkName)
```

Comment:
Can we keep all this meta logic - isFathomNetworkName, and isSmartContractWallet as a part of the `trackEvent` function? Will greatly simplify the redundant code.
The new contract for trackEvent can be: 
`trackEvent(event, additionalParams)` where `additionalParams` can contain `l2NetworkName` and `isSmartContractWallet` etc, and we can easily maintain the logic to send the event to Fathom conditionally at a central level and not every time before calling trackEvent.

Reply:
Even if we pass `l2NetworkName` and `isSmartContractWallet` as `additionalParams` props we will have to make checks inside `trackEvent` to see if specific event was called and only then apply `additionalParams` as conditions. This could be a larger restructure to the code.

Right now we make these checks in the `transfer` method but that covers 10+ events.

What we could do is to have another prop in `trackEvent`, something like `conditions` of type `boolean[]`. We would still have to pass conditional checks in there but we don't have to wrap our code in `if` statements and it's more scalable if we wanted to call events conditionally in the future.

E.g. use case: `trackEvent(event_name, [!isSmartContractWallet, isFathomNetwork(l2NetworkName)]`.
And then in `trackEvent`: `const isValid = conditions.every(c => c === true)`.

Thoughts on all that?

Reply:
As per our conversation, we don’t have a requirement to limit all the events to specific conditions (like no testnet, no smart contract wallet) at a generic level. These conditions are event specific and hence make sense to be outside of trackEvent code.
Such requirements might come later but it will require a much better refactoring then. For now the current approach makes sense.